### ISSUE TITLE
[BUG] Postgres functions wrong formatted

### ISSUE THREAD

**Issue by KirioXX** [cweijan/vscode-mysql#748]
Hi,

We are using [supabase](https://supabase.com/) for our backend service.
For that, we rely heavily on Database Functions but every time I open a function it is miss formatting the function.
Here an example:
```sql
-- Active: 1667981329659@@127.0.0.1@54332@postgres@store
CREATE OR REPLACE FUNCTION store.generate_emoji()
 RETURNS text
 LANGUAGE plpgsql
AS $
    DECLARE
        emojis          text = '🐍🐐🐝🐇🐢🐠🐘🐒🐬🐊🐺🐤🐎🐦🐋🐛🐔🐟🐕🐑🐧🐙🐩🐓🐏🐅🐉🐨🕷🐞🐈🦀🐀🐂🐗🐜🦂🐄🐿🏇🐖🐆🐪🐡🐁🐃🦅🦆🦇🦈🦉🦋🦌🦍🦎🦏🦐🦒🦓🦔🦖🦘🦙🦚🦛🦜🦝🦞🦡🦢🐿️🦦';
    BEGIN
       return substr(emojis, (floor(random() * (length(emojis) - 1 + 1)) + 1)::int, 1);
    END;
$
```
This throws an error that there is no closing $.

In pgAdmin4 this function would look like this:
```sql
CREATE OR REPLACE FUNCTION store.generate_emoji(
	)
    RETURNS text
    LANGUAGE 'plpgsql'
    COST 100
    VOLATILE PARALLEL UNSAFE
AS $BODY$
    DECLARE
        emojis          text = '🐍🐐🐝🐇🐢🐠🐘🐒🐬🐊🐺🐤🐎🐦🐋🐛🐔🐟🐕🐑🐧🐙🐩🐓🐏🐅🐉🐨🕷🐞🐈🦀🐀🐂🐗🐜🦂🐄🐿🏇🐖🐆🐪🐡🐁🐃🦅🦆🦇🦈🦉🦋🦌🦍🦎🦏🦐🦒🦓🦔🦖🦘🦙🦚🦛🦜🦝🦞🦡🦢🐿️🦦';
    BEGIN
       return substr(emojis, (floor(random() * (length(emojis) - 1 + 1)) + 1)::int, 1);
    END;
$BODY$;
```

It becomes even worse in a [plv8](https://plv8.github.io/) function because it strips off all arguments in the [plv8.execute](https://plv8.github.io/#-code-plv8-execute-code-) functions.

Thanks!

- Operating system: Docker / [Supabase CLI](https://supabase.com/docs/guides/resources/supabase-cli)
- Database type and version: PostgresSQL 15.1

**Comment by cweijan** [comment]
Thanks for the feedback, I'll try to fix it.

**Comment by KirioXX** [comment]
Awesome! Thanks @cweijan 🙌

**Comment by cweijan** [comment]
Fix at version 6.1.5.
